### PR TITLE
Add prototype code for gmail account creation

### PR DESCRIPTION
**Pull request by helenst**
Needs to be integrated into something! Also relies on a credentials json file which must be separately supplied.

### CODE REVIEW

File: prototype/gmail_accounts.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+from random import choice
+
+from apiclient.errors import HttpError
+from apiclient.discovery import build
+from oauth2client.service_account import ServiceAccountCredentials
+
+
+PASSWORD_CHARS = string.ascii_letters + string.digits
+
+def generate_password(length=10):
```
Comment (bmispelon) [inline prototype/gmail_accounts.py]: There's a `get_random_string` function in `django.utils.crypto`, maybe we can reuse that?

Comment (helenst) [inline prototype/gmail_accounts.py]: Did not know about that. Sounds like a good idea - thanks! :+1:

File: prototype/gmail_accounts.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    # The old email address is kept as an alias to the new one, but we don't want this.
+    service.users().aliases().delete(userKey=new_email, alias=old_email).execute()
+
+    print('Account %s renamed to %s' % (old_email, new_email))
+
+    return response
+
+
+def get_account(slug):
```
Comment (bmispelon) [inline prototype/gmail_accounts.py]: What about calling this function `print_account` instead?

File: prototype/gmail_accounts.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+2. Create a service account to enable 2 legged oauth (https://developers.google.com/identity/protocols/OAuth2ServiceAccount)
+3. Enable delegation of domain-wide authority for the service account.
+4. Enable Admin SDK for the domain.
+5. Give the service account permission to access admin.directory.users service (https://admin.google.com/AdminHome?chromeless=1#OGX:ManageOauthClients).
+"""
+
+SCOPES = 'https://www.googleapis.com/auth/admin.directory.user'
+
+credentials = ServiceAccountCredentials.from_json_keyfile_name(
+    'djangogirls_website-93134565f52d.json',
```
Comment (olasitarska) [inline prototype/gmail_accounts.py]: I understand that this file will need to be deployed to pythonanywhere?

Comment (helenst) [inline prototype/gmail_accounts.py]: Yes - I guess it will need to be put into place manually.

File: prototype/gmail_accounts.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+def create_account(slug, city_name):
+    """
+    Create a new account
+
+    e.g. create_account('testcity', 'Test City')
+    """
+    email = make_email(slug)
+    password = generate_password()
+    response = service.users().insert(body={
```
Comment (aniav) [inline prototype/gmail_accounts.py]: What happens if a specified e-mail already exists? We should be able to catch the error and, for example, display it in the admin as a ValidationError

File: prototype/gmail_accounts.py (hunk 3 above)
Comment (olasitarska) [inline prototype/gmail_accounts.py]: Make sense :)

File: prototype/gmail_accounts.py (hunk 4 above)
Comment (helenst) [inline prototype/gmail_accounts.py]: It'll raise an HttpError (with code 409). I'll document that in the comment for now.

Comment (aniav) [inline prototype/gmail_accounts.py]: Ah, great! 💃

**Pull request merged**